### ISSUE TITLE
Repacking doesn't work

### ISSUE THREAD

**Issue by SillyconWilly** [adm244/AGSUnpacker#5]
Hi!

You write in the instructions that you can also use your tool for repacking. Unfortunately, the repack assets button cannot be clicked. Only the repack sprites can be clicked. Can you please tell me how I can repack the edited files?

bye

**Comment by adm244** [comment]
Yes, assets repackaging are not yet implemented.

In the meantime, you can store raw assets in root game folder and engine should prefer them over packaged ones.
Or use another tool like agsutils.

**Comment by SillyconWilly** [comment]
Hi!

Okay, thank you for that hint. :-)
bye

    Am Montag, 30. Dezember 2024 um 23:35:04 MEZ hat Adam Worgen ***@***.***> Folgendes geschrieben:   

 


Yes, assets repackaging are not yet implemented.

In the meantime, you can store raw assets in root game folder and engine should prefer them over packaged ones.
Or use another tool like agsutils.

**Comment by TakutoRU** [comment]
"In the meantime, you can store raw assets in root game folder and engine should prefer them over packaged ones."
Hello. Could you please give some info, how paths should be organised, for game engine found and loaded raw png sprites over those in acsprset.spr file? Is it possible to partially replace sprites, for localisation purproses?

**Comment by adm244** [comment]
> Is it possible to partially replace sprites

No, AGS engine doesn't allow that (at least at a moment). It can only replace files that are stored in `*.ags` archives (i.e. `.spr`, `.ttf`, `.wfn`, `.crm`, `.dta`, `.dat` etc.) It cannot replace some parts of those files, only an entire file (it technically can replace certain script files, but it's implemented in such a way that it has no practical use).

Many games don't compress sprites file by default, so it's possible to decrease its size by enabling either LZW or Deflate (preferred).

> Could you please give some info, how paths should be organised

If you place files like `room1.crm`, `agfnt1.ttf`, `acsprset.spr` or `sprindex.dat` in root game folder (where `.ags` or `.exe` is located) then AGS will use these files instead of original ones.

**Comment by TakutoRU** [comment]
Thanks for info. Its a pity, seems like only way to distribute fan translation is to put all sprites with it, even those which not need to be localized (
| Many games don't compress sprites file by default, so it's possible to decrease its size by enabling either LZW or Deflate (preferred)
Do you mean to compress every png sprite with image converter? Or such compression is part of spr file packing software?
Game which Im trying to localize has enormous spr file, almost 2 gigs when packing with AGSUnpacker, though all files unpacked only 400 megs. Maybe text only localization then (
| If you place files like room1.crm
So, rooms can be ovverriden individually?

**Comment by adm244** [comment]
> Do you mean to compress every png sprite with image converter? Or such compression is part of spr file packing software?

It's part of `spr` file. The reason why unpacked sprites size is only 400 MB is because AGSUnpacker saves images as `png` files which use deflate compression. When these sprites are stored in `spr` file they might not be compressed.

Compression for `spr` file can be "turned on" by modifying 3rd byte in `header.bin` file. If it's 0 then compression is disabled. To enable deflate compression set this byte to 3. Engines older than 3.6.1 doesn't support deflate and you could try LZW method by setting value to 2.

> So, rooms can be ovverriden individually?

Yes.

**Comment by TakutoRU** [comment]
Thanks again, and please sorry for off-topic discussions.
Seems like my game on AGS 3.4 and crashes on sprites compressed with Deflate algo.
When set to LZW, AGSUnpacker seemingly doing something very slow (9% CPU utilization on Ryzen 5700x) for at least an hour, and without result in this time (0 bytes file). If force closed, some sprite file created, but probably truncated - game crashes. Now I'll try to wait for process to end normally, leave program for the night, but for repeatable test process its unusable (

**Comment by adm244** [comment]
> When set to LZW, AGSUnpacker seemingly doing something very slow (9% CPU utilization on Ryzen 5700x) for at least an hour, and without result in this time (0 bytes file)

This is definitely a bug. I'll take a look.

**Comment by TakutoRU** [comment]
Result - process takes out

<img width="413" height="228" alt="Image" src="https://github.com/user-attachments/assets/636c56cf-5c84-4ad1-b031-0359c6b59b69" />

 around 1.5 hours, result is 630 Mb file, game crashes anyways with message on screen.
